### PR TITLE
To do list

### PR DESCRIPTION
**Pull request by karam084**
* Project requirements

- Set up a new project with webpack that is based on the webpack exercise you have already completed.
- Create an index.html file and write your HTML markup here. Create an empty To Do List placeholder (<div> or <ul> element). The index.html file must be set as a template using the HTML Webpack Plugin.
- Create an index.js file and set an array of some simple to do tasks (array of objects). Each task object should contain three keys:
- description [string].
- completed [bool].
- index: [number].
- Write a function to iterate over the tasks array and populate an HTML list item element for each task.
- On page load render the dynamically created list of tasks in the dedicated placeholder. The list should appear in order of the index values for each task.
- Create a style.css and set rules for the To Do List. CSS must be loaded by Webpack Style/CSS Loader. Your list should be a clone of the part of the minimalist project captured in the video below.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import './style.css';
+import './style.scss';
+import 'bootstrap';
+
+const toDolist = document.getElementById('toDoList');
+const addBtn = document.getElementById('addBtn');
+const inputs = document.getElementsByClassName('forminput');
+let listArray = [];
+
```
Comment (TedLivist) [inline src/index.js]: - [x] You should have an array containing some tasks. Each task should have `description` (string), `completed` (boolean) and `index` (number) properties. Kindly add this.

  See example below
  ```
  const listArray = [
    {
      description: 'First task',
      completed: false,
      index: 1
    },
    {
      description: 'Second task',
      completed: true,
      index: 2
    },
    {
      description: 'Final task',
      completed: false,
      index: 3
    },
  ]
  ```

File: README.md [hunk 2]
```diff
@@ -4,10 +4,7 @@
```
Comment (TedLivist) [inline README.md]: - [x] In your README, kindly change the title from the generic **`Project Name`** to something more descriptive of the project you just built.

  karam-proj

- [x] In your README, kindly add detailed instructions on how users can set up and run the app locally i.e. from cloning the repo to running `npm install` (for installing packages) to running the `npm start` (to start the app)

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  let trs = '';
+  for (let i = 0; i < listArray.length; i += 1) {
+    trs += `<tr>
+     <td><input type="checkbox"></td>
+    <td class="float-left">${listArray[i].name}</td>
+    <td><button onclick="deleteList(${i})" class="float-right"><i class="fa fa-ellipsis-v"></i></button></td>
+    </tr>`;
+  }
+  document.getElementById('tableBody').innerHTML = trs;
+}
```
Comment (TedLivist) [inline src/index.js]: - [x] You have a function that renders the array. That's good. However, you need to ensure your design is similar to what we have on the template shared with you.

  The template | Your design
  :---:|:---:
  karam-todo-template | karam-todo-implement

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @karam084,

Good job so far!
✔️ App is set up with Webpack
✔️ There is function for displaying tasks
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+function displayData() {
+  let trs = '';
+  for (let i = 0; i < listArray.length; i += 1) {
+    trs += `<tr>
+     <td><input type="checkbox"></td>
+    <td class="float-left">${listArray[i].name}</td>
+    <td><button onclick="deleteList(${i})" class="float-right mt-lg-1"><i class="fa fa-ellipsis-v"></i></button></td>
+    </tr>`;
+  }
+  document.getElementById('tableBody').innerHTML = trs;
```
Comment (BenMukebo) [inline src/index.js]: - [ ] Please make sure you render properly your array items to the front end, actually, it returns `unddefined`
arry

**Review by BenMukebo — CHANGES_REQUESTED** [review]
## Hi **@karam084,**

You did a great job so far on making changes as requested y the previous reviewer 👏🏼 👏🏼 but
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💪🏻

- No linter errors. ✔️ 
- Good commit messages ✔️ 
- Professional README ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏 💻  

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AshakaE** [comment]
Hi @karam084 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**